### ISSUE TITLE
Multiple language support

### ISSUE THREAD

**Issue by alexandreleroux** [newmarcel/KeepingYouAwake#20]
Minor feature request. But first, thanks a lot for this valuable project :-)

If you implement multiple language support for KYA's menu items, I'll gladly contribute the French translation. While a minor improvement, multiple language support does matter to some users. Thanks for the great (and open source) software -- Alex

**Comment by iberezovsky** [comment]
I can also help with Russian translation;)

**Comment by newmarcel** [comment]
Cool thanks! I think the app will be ready for localisations after I finished the next big update v1.3 *(I don't know yet when it will be finished)*.

I have to think about a method to provide you the string tables. And I will add a German translation as well. :smiley: 

I'll let you know when I'm ready.

**Comment by sebthedev** [comment]
Hi @newmarcel, thanks for the great app!

I am a Mac developer myself and I have found that [this](http://xliff.brightec.co.uk) free online .xliff editor is a great tool for translators to use to translate apps. Simply export your .xliff file from Xcode (Editor > Export for Localisation…) then get the translators to upload to the editor.

All the best,
Sebastian

**Comment by spaceisstrange** [comment]
Any news on this? I can help with the spanish translation :smile:

**Comment by newmarcel** [comment]
I currently plan to release version 1.4.0 first before allowing translations. After that, I will start fixing some areas of the app which are not l10n-friendly.

With the first beta of version 1.4.1 I will provide the current `XLIFF` files in the project repository, a basic localization guide in the wiki and I will also maintain the German localization as a template for other languages. New translations could then be added by pull request.

I currently have two issues with providing translations:

##### 1) How can I verify that a translation is correct?

Should at least a second native speaker approve a translation before it is allowed into the app? Other ideas?

##### 2) How should I deal with incomplete translations in the future?

Let's say I have a complete set of *Klingon* translations for 1.4.1, I update many areas of the app for the next major release, but I can't reach the translator anymore… should I just remove the translation? Or should I hold back the release until I find a new translator?

I'm thankful and very much open for feedback from the community regarding this process :+1:

**Comment by thegcat** [comment]
From what I know from other Open Source projects: 

1. Accept whatever translation is proposed as-is, anything more (double checks, authentication to make sure it's the correct contributor, and so on) will discourage some people. People mostly want to do the right thing, and a grammatically not-perfect translation is better than none at all.
1. Have a base locale, in this case probably english, and fall back to that for untranslated strings. 80% translated with a few obscure options in english is better than not translated at all.

I think it should be kept in mind that this is a voluntary and group effort, i.e. not to expect perfection but still welcome everyone wanting to contribute.

One last thing: I'm not sure how this is usually handled in the Mac world, but there's a few web services out there offering very compelling interfaces to create and maintain translation files, this might be worth investigating.

**Comment by timbru31** [comment]
Hi there,

I've worked with [Transifex](https://www.transifex.com/) in the past and it worked well. You can add the string that needs to be translated, multiple people can edit the same language and/or re-approve.
Another platform I know, but haven't used, is [Crowdin](https://crowdin.com/) (Free for Open Source)

Regarding the language approval, even big companies like Mojang had issues with that, since some people added offenses to community driven translation project. Many other projects are fine with user submitted localizations and have trusted the users.
On the other hand, WhatsApp for example has a voting feature for new translations.

Regarding the second point, I normally experience the following: If a translation string is missing, there is a default (in most cases English), until the locallization was uptaed. A missing translation string shouldn't keep you back from pushing a new release out IMHO.

To sum it up, I'd appreciate the translation approach.

**Comment by alexandreleroux** [comment]
@newmarcel : Should we close this ticket now that localization is supported by KYA?

@iberezovsky : Don't hesitate to contribute a russian translation, see https://github.com/newmarcel/KeepingYouAwake/tree/master/Localizations/KeepingYouAwake

Cheers -- Alex

**Comment by newmarcel** [comment]
@alexandreleroux Yes, I will close this ticket. We have a great set of localizations now 🎉
